src/pfe/portal/modules/LoadRunner.js

```diff
@@ -487,91 +505,99 @@ module.exports = class LoadRunner {
     * Socket event for connect
     * Logs connection success
     */
-    this.socket.on('connect', () => {
-      log.info('Loadrunner has connected')
-      this.up = true;
+    this.socket.on('connect', data => {
+      //if (data.projectID === this.project.projectID) {
+        log.info('Loadrunner has connected')
+        this.up = true;
+      //}
     });
 
     /**
     * Socket event for disconnect
     * Logs the error and starts reconnect procedure
     */
-    this.socket.on('disconnect', () => {
-      log.info('Loadrunner has disconnected')
-      // If this.up is false we're already trying to reconnect
-      if (this.up) {
-        // socket.io-client will automatically reconnect and trigger the connect event.
-        this.up = false;
+    this.socket.on('disconnect', data => {
+      if (data.projectID === this.project.projectID) {
+        log.info('Loadrunner has disconnected')
+        // If this.up is false we're already trying to reconnect
+        if (this.up) {
+          // socket.io-client will automatically reconnect and trigger the connect event.
+          this.up = false;
+        }
       }
     });
 
     /**
     * Socket event for connection timeout
     * Logs the error and starts reconnect procedure
     */
-    this.socket.on('connect_timeout', () => {
-      log.info('Loadrunner has timed out')
-      // If this.up is false we're already trying to reconnect
-      if (this.up) {
-        // socket.io-client will automatically reconnect and trigger the connect event.
-        this.up = false;
-      }
+    this.socket.on('connect_timeout', data => {
+      if (data.projectID === this.project.projectID) {
+        log.info('Loadrunner has timed out')
+        // If this.up is false we're already trying to reconnect
+        if (this.up) {
+          // socket.io-client will automatically reconnect and trigger the connect event.
+          this.up = false;
+        }
+      }  
     });
 
     /**
     * Socket event for error
     * Logs the error and starts reconnect procedure
     */
-    this.socket.on('error', (err) => {
-      log.error('LoadRunner socket error');
-      // A socket error could occur while a load run is not in progress.
-      if (this.project) {
-        this.project.loadInProgress = false;
-      }
-      log.error(err);
-      // If this.up is false we're already trying to reconnect
-      if (this.up) {
-        // socket.io-client will automatically reconnect and trigger the connect event.
-        this.up = false;
+    this.socket.on('error', (err, data) => {
+      if (data.projectID === this.project.projectID) {
+        log.error('LoadRunner socket error');
+        // A socket error could occur while a load run is not in progress.
+        if (this.project) {
```

Comment:
I think in the new model this might be wrong. I can't see anywhere that checks `loadInProgress` on the project object anymore (it's not used by loadtest.route.js in particular( so I think this needs to be changed to mark the load run (if running) as finished the right way.

Reply:
Yep, that's a mistake. Will update to use the ifIdle() method instead, as everything else is using it now